game-backend/src/MainController.ts

```diff
@@ -48,26 +54,65 @@ export default class MainController {
     matchUpdate: MatchUpdate
   ) => void {
     return (userId: UserId, matchUpdate: MatchUpdate): void => {
-      this.store.userContexts.userContextMap[userId].forEach(ws => {
+      this.store.userConnnections.userConnectionMap[userId].forEach(ws => {
         ws.send(JSON.stringify(matchUpdate));
       });
     };
   }
 
   private getTerminateExecutor(): () => void {
     return (): void => {
+      // ping all websocket connections
+      this.pingAllConnections();
       setTimeout(() => {
+        this.removeInactiveUsers();
+
         this.gameController = new GameController(
           this.getMatchUpdateExecutor(),
           this.getTerminateExecutor()
         );
-        this.store.userContexts.userContextIds.forEach(id => {
+        this.store.userConnnections.userConnectionIds.forEach(id => {
           this.gameController.appendUser(id);
         });
       }, GAME_RESTART_TIME);
     };
   }
 
+  private pingAllConnections(): void {
```

Comment:
This would probably become obsolete then.

Reply:
I thought about it. We could scratch either the regular checking or this call. It depends on what we want to achieve. I implemented that because I wasn't sure how often we want to remove stale connections. Because if the heartbeat interval is greater than the game restart time it wouldn't work. 
But I think we can remove the calls for now, if we agree on the < 10s heartbeat interval.